runtime/interface.go

```diff
@@ -41,93 +43,124 @@ type Block struct {
+	// StorageUsed gets storage used in bytes by the address at the moment of the function call.
+	StorageUsed(address Address, caller Location) (value uint64, err error)
+	// Note: StorageCapacity has been moved to injected methods (similar to get balance)
 
-	// SetCadenceValue sets a value for the given key in the storage, owned by the given account.
-	SetCadenceValue(owner Address, key string, value cadence.Value) (err error)
+	// AddAccountKey appends a key to an account.
+	AddAccountKey(address Address, publicKey []byte, caller Location) error
+	// RemoveAccountKey removes a key from an account by index.
+	RevokeAccountKey(address Address, index uint, caller Location) error
+	// AccountPublicKey returns the account key for the given index
+	AccountPublicKey(address Address, index uint, caller Location) (publicKey []byte, err error)
+	// VerifyAccountSignature verifies a signature for the given address and index
+	// TODO RAMTIN do I need the tag here?
+	VerifyAccountSignature(address Address, index uint, signature []byte, tag string, signedData []byte, caller Location) (isValid bool, err error)
+}
+
+// Results are responsible to capture artifacts generated
+// when running the runnable
+//
+// Results won't be directly callable by users
+// and will be used by the runtime to capture outputs
+type Results interface {
+	// AppendLog appends a log to the log collection
+	AppendLog(string) error
+	// Logs returns all the logs
+	Logs() ([]string, error)
+	// returns log i of the log collection
+	Log(i uint) (string, error)
+
+	// AppendEvent appends an event to the event collection
+	AppendEvent(cadence.Event) error
+	// Events returns all the events
+	Events() ([]cadence.Event, error)
+	// returns event i of the event collection
+	Event(i uint) (cadence.Event, error)
+
+	// AppendError appends a non-fatal error
+	AppendError(error)
+	// return all the errors
+	Errors() multierror.Error
+
+	// AddComputationUsed adds a new uint64 value to the computationUsed (computation accumulator)
+	AddComputationUsed(uint64)
+	// ComputationSpent returns the total amount of computation spent during the execution
+	ComputationSpent() uint64
 }
 
 type Metrics interface {
+	// ProgramParsed captures the time spent on parsing the program
 	ProgramParsed(location common.Location, duration time.Duration)
+	// ProgramChecked captures the time spent on checking the parsed program
 	ProgramChecked(location common.Location, duration time.Duration)
+	// ProgramInterpreted captures the time spent on interpreting the parsed and checked program
 	ProgramInterpreted(location common.Location, duration time.Duration)
+
+	// ValueEncoded captures the time spent on encoding a value
+	// TODO: maybe add type
 	ValueEncoded(duration time.Duration)
+	// ValueDecoded capture the time spent on decoding an encoded value
+	// TODO: maybe add type
 	ValueDecoded(duration time.Duration)
 }
 
+type HighLevelAccounts interface {
+	Accounts
```

Comment:
Would be nice to just have this as `HighLevelStorage` and not embed any other interface, it's stand-alone